src/config.rs

```diff
@@ -21,58 +123,59 @@ struct JsonEntry {
     description: String,
 }
 
-/// StorageAdapter is the interface for storing and retrieving settings
-/// This can be used to store settings in a database, json file, etc
-pub trait Store {
-    /// Retrieves a setting from the storage
-    fn get_setting(&self, key: &str) -> Option<Setting>;
-    /// Stores a given setting to the storage
-    fn set_setting(&mut self, key: &str, value: Setting);
-    /// Retrieves all the settings in the storage in one go. This is used for preloading the settings
-    /// into the ConfigStore and is more performant normally than calling get_setting manually for each
-    /// setting.
-    fn get_all_settings(&self) -> Result<HashMap<String, Setting>>;
-}
-
-/// Configuration store is the place where the gosub engine can find all configurable options
+/// Configuration storage is the place where the gosub engine can find all configurable options
 pub struct ConfigStore {
     /// A hashmap of all settings so we can search o(1) time
-    settings: HashMap<String, Setting>,
+    /// The mutex allows to share between multiple threads,
+    /// The refcell allows us to use mutable references in a non-mutable way (ie: settings can be
+    /// stored while doing a immutable get())
+    settings: std::sync::Mutex<std::cell::RefCell<HashMap<String, Setting>>>,
```

Comment:
A mutex to share the config store between threads
A RefCell to allow us to mutate from a immutable config store